url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/GUI.java

```diff
@@ -0,0 +1,598 @@
+package net.nexon.vindictus.itemcomparer;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.EventQueue;
+import java.awt.Font;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.File;
+
+import javax.swing.AbstractListModel;
+import javax.swing.BoxLayout;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JSplitPane;
+import javax.swing.JTextField;
+import javax.swing.ListSelectionModel;
+import javax.swing.SwingConstants;
+import javax.swing.UIManager;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.TitledBorder;
+import javax.swing.JSlider;
+
+import net.nexon.vindictus.itemcomparer.inport.StaticScrolls;
+import javax.swing.JComboBox;
+import javax.swing.DefaultComboBoxModel;
+import javax.swing.ButtonGroup;
+import java.awt.GridLayout;
+import java.awt.FlowLayout;
+
+public class GUI extends JFrame {
+
+	private JPanel contentPane;
+
+	private final static String VERSION = Main.VERSION;
+	private final int bitchcount = 6;
+
+	private JTextField minatk_Text;
+	private JTextField ignore_Text;
+	private JTextField result_Text;
+	private JTextField price_Text;
+	private JList char_List;
+	private JList sort_List;
+	private JList pre_List;
+	private JList suf_List;
+	private JList minlvl_List;
+	private JList maxlvl_List;
+	private JList smin_List;
+	private JList smax_List;
+	private JList cpu_List;
+	private JCheckBox noinfo_Check;
+	private JCheckBox nocsv_Check;
+	private JCheckBox regina_Check;
+	private JSlider plus_Slider;
+	private JTextField minmatk_Text;
+	private JComboBox apreBox = new JComboBox();
+	private JComboBox asufBox = new JComboBox();
+	private final ButtonGroup buttonGroup = new ButtonGroup();
+	private JCheckBox barha_Check = new JCheckBox("barha weapon");
+	private JCheckBox guard_Check;
+	private JCheckBox cloth90_Check;
+	/**
+	 * Launch the application.
+	 */
+	public static void main(String[] args) {
+		EventQueue.invokeLater(new Runnable() {
+			public void run() {
+				try {
+					GUI frame = new GUI();
+					frame.setVisible(true);
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+		});
+	}
+
+	private void calc() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 135-181 into a separate method.